packages/core/src/discovery.js

```diff
@@ -79,7 +81,18 @@ export function createRequestFinishedHandler(network, {
           return log.debug(`- Skipping disallowed resource type [${request.type}]`, meta);
         }
 
-        resource = createResource(url, body, response.mimeType, {
+        // Ensure we're getting a proper mimetype if the server or
+        // asset discovery browser is sending `text/plain`
+        let resourceMimeType = response.mimeType === 'text/plain' ? mime.lookup(response.url) : response.mimeType;
```

Comment:
This could be a boolean `false` if a URL without an extension has a `text/plain` mimetype.

```suggestion
        // a text/plain mimetype could be a fallback, try to infer one from the response url
        let mimeType = (url.response.mimeType === 'text/plain' && mime.lookup(response.url)) || response.mimeType;
```

Changing it from a ternary will ensure that `response.mimeType` is used if `mime.lookup()` returns `false`.